ch4/deepComparison.js

```diff
@@ -0,0 +1,42 @@
+const deepEqual = (value1, value2) => {
```

Comment:
you should add some more complex test cases for this function. specifically, test arrays, objects, objects with nested arrays, arrays with nested objects, etc.

for example, off the top of my head, this implementation doesn't work for:
- `{a: 1}` and `{a: 1, a: 2}`
- `{0: 1, 1: 2}` and `[1, 2]`
- `{}` and `[]`
- `[undefined]` and `[null]`

there are [a bunch of weird edge cases](https://github.com/denysdovhan/wtfjs?tab=readme-ov-file#-examples), but we won't worry about those here. don't worry about `NaN` either, but i'd say `null` and `undefined` are fair game.

Reply:
roger

Reply:
done